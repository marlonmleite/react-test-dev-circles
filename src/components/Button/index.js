import styled, { css } from 'styled-components'

const getPrimaryStyle = (props) => props.primary && css`
  background: #61DAFB;
  color: #282C34;
`

const Button = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 2px solid #61DAFB;
  color: #61DAFB;
  font-weight: bold;
  margin: 0.5em 1em;
  padding: 0.25em 1em;

  ${getPrimaryStyle}

  :hover {
    cursor: pointer;
  }
`;

export default Button;

/*<style>
  .my-button {
    background: transparent;
    border-radius: 3px;
    border: 2px solid palevioletred;
    color: palevioletred;
    margin: 0.5em 1em;
    padding: 0.25em 1em;
  }

  .my-button-primary {
    background: palevioletred;
    color: white;
  }
</style>

<button class="my-button my-button-primary" />*/