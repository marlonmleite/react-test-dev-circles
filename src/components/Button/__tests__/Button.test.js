import React from 'react'
import { shallow } from 'enzyme'
import Button from '../'

test('should render', () => {
  const button = shallow(<Button>btn</Button>)

  expect(button).toMatchSnapshot()
})

test('should handle click', () => {
  const onClick = jest.fn()
  const button = shallow(<Button onClick={onClick}>btn</Button>)

  button.simulate('click')

  expect(onClick).toHaveBeenCalled()
})

/*test.skip('should render button primary', () => {
  const button = shallow(<Button primary>primary</Button>)

  expect(button).toHaveStyleRule('background', '#61DAFB')
  expect(button).toHaveStyleRule('color', '#282C34')
})*/

test('should hover button', () => {
  const button = shallow(<Button>Save</Button>)

  expect(button).toHaveStyleRule('cursor', 'pointer', {
    modifier: ':hover'
  })
})

/*test('should handle change', () => {
  const event = { target: { value: 'asdf' } }
  const onChange = jest.fn()
  const button = shallow(<input onChange={onChange}></input>)

  button.simulate('change', event)

    expect(onChange).toHaveBeenCalledWith(event)
})*/