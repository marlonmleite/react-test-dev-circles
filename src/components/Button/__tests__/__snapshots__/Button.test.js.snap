// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.c0 {
  background: transparent;
  border-radius: 3px;
  border: 2px solid #61DAFB;
  color: #61DAFB;
  font-weight: bold;
  margin: 0.5em 1em;
  padding: 0.25em 1em;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
>
  btn
</button>
`;
